kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/Utils.java

```diff
@@ -124,4 +126,48 @@ public static void closeCloseablesSilently(AutoCloseable... closeables) {
       // ignore
     }
   }
+
+  // Utility class to support `intoRows` below
+  private static class FilteredBatchToRowIter implements CloseableIterator<Row> {
```

Comment:
Sorry I think maybe I wasn't clear; I meant we should add a `flatMap` util to the `CloseableIterator` interface and we can generally use that for everything else. Fine to do this in a follow-up though don't want to block on this